test/new_relic/agent/distributed_tracing/trace_context_cross_agent_test.rb

```diff
@@ -253,10 +253,14 @@ def verify_attributes(test_case_attributes, actual_attributes, event_type)
               %Q(Unexpected #{event_type} attribute "#{key}")
           end
 
-          (test_case_attributes['vendors'] || []).each do |k, v|
-            assert_equal v,
-              actual_attributes[k.to_s],
-              %Q(Wrong "#{k}" #{event_type} attribute; expected #{v.inspect}, was #{actual_attributes[k.to_s].inspect})
+          test_key = 'tracingVendors'
+          actual_key = "tracestate.#{test_key}"
+
+          if test_case_attributes.key?(test_key)
```

Comment:
Why did you decide to wrap this test case in an if statement?

Reply:
I had the same question. The specific new "vendor" functionality went in at the bottom of a giant method that is being used to test lots of things for ALL of the JSON payloads. There's already a pattern in place there for having each new group of assertions skip any irrelevant payload and Hannah matched the existing style.